app/parse.py

```diff
@@ -8,9 +19,137 @@
+
+
+def get_single_page_quotes(page_soup: Tag) -> [Quote]:
+    quotes = page_soup.select(".quote")
+    return [parse_single_quote(quote_soup) for quote_soup in quotes]
+
+
+def get_all_pages_quotes() -> [Quote]:
+    # logging
+    time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+    logging.info(f"{time} - Start parsing quotes")
+
+    page_text = requests.get(BASE_URL).content
+    first_page_soup = BeautifulSoup(page_text, "html.parser")
+
+    all_quotes = get_single_page_quotes(first_page_soup)
+
+    # next button searching
+    next_element = first_page_soup.select_one(".next")
+    next_page_number = next_element.a["href"].split("/")[-2]
+    while next_element is not None:
+        try:
+            # logging
+            time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+            logging.info(f"{time} - Start parsing page #{next_page_number}")
+
+            next_page_url = urljoin(BASE_URL, f"/page/{next_page_number}/")
+            next_page_text = requests.get(next_page_url).content
+            next_page_soup = BeautifulSoup(next_page_text, "html.parser")
+
+            # Adding quotes from a new page
+            all_quotes.extend(get_single_page_quotes(next_page_soup))
+            # all_authors.extend(get_single_page_quotes(next_page_soup)[1])
+
+            # next button searching
+            next_element = next_page_soup.select_one(".next")
+            if next_element is None:
+                raise AttributeError("Element '.next' not found")
+            next_page_number = next_element.a["href"].split("/")[-2]
+        except AttributeError as e:
+            time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+            logging.info(f"{time} - 'Next' button not found")
+            break
+    return all_quotes
+
+
+def write_quotes_to_csv(output_csv_path: str) -> None:
+    quotes = get_all_pages_quotes()
+    with open(output_csv_path, "w", encoding="utf-8") as file:
+        writer = csv.writer(file)
+        writer.writerow(QUOTE_FIELDS)
+        writer.writerows([astuple(quote) for quote in quotes])
+
+
+def format_name(input_name: str) -> str:
+    # Removing diacritics
+    normalized_name = unicodedata.normalize("NFD", input_name)
+    name_without_accents = "".join(
+        char for char in normalized_name if unicodedata.category(char) != "Mn"
+    )
+    # Replace spaces and dots with hyphens
+    formatted_name = re.sub(
+            r"[.\s]+", "-", name_without_accents
+        ).replace("'", "").rstrip("-")
+    return formatted_name
+
+
+def get_all_authors_biography(authors: list) -> [Author]:
+    returning_authors_list = []
+    for name in authors:
+        refactored_name = format_name(name)
+        author_url = urljoin(BASE_URL, f"/author/{refactored_name}/")
+        page_text = requests.get(author_url).content
+        first_page_soup = BeautifulSoup(page_text, "html.parser")
+        biography = first_page_soup.select_one(
+            ".author-description"
+        ).text.strip()
```

Comment:
The code assumes that `.author-description` will always be present. If it's missing, this will raise an error. Consider adding a check to handle cases where the element is not found.